File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
    let obj = {};
    for(const val of arr) {
        obj[val] = (obj[val] || 0) + 1;
    }
    const output = Object.values(obj);
    return output.length == new Set(output).size
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        count_dict = {}
        for num in arr:
            count_dict[num] = count_dict.get(num, 0) + 1
                
        results = len(count_dict.values()) == len(set(count_dict.values()))
        
        return results